### PR TITLE
github: Add Doxyfile and github doxygen action

### PR DESCRIPTION
**Pull request by drakenclimber**
Add a default Doxyfile and a doxygen action.

Signed-off-by: Tom Hromatka <tom.hromatka@oracle.com>

### CODE REVIEW

**Comment by drakenclimber** [comment]
If we decide we like this, I will add a link to the Doxygen on the libcgroup README.  I don't want to add that now because the branch, `doxygen/main`, won't exist until this is committed.

File: .github/workflows/continuous-integration.yml [hunk 1]
```diff
@@ -34,6 +34,27 @@ jobs:
...
+        submodules: false
+    - uses: mattnotmitt/doxygen-action@v1
+      with:
+        working-directory: "./"
+        doxyfile-path: "./doc/Doxygen"
+    - name: Deploy Doxygen
+      uses: peaceiris/actions-gh-pages@v3
+      with:
+        github_token: ${{ secrets.GITHUB_TOKEN }}
+        publish_dir: ./html
```
Comment (kamalesh-babulal) [inline .github/workflows/continuous-integration.yml:55]: That's cool, thanks for working on it. A minor nitpick, should it be deployed under `./doc/html` ?

Comment (drakenclimber) [reply .github/workflows/continuous-integration.yml:55]: Good call.  Thanks!

**Review by kamalesh-babulal — APPROVED** [review]

**Comment by kamalesh-babulal** [comment]
Reviewed-by: Kamalesh Babulal <kamalesh.babulal@oracle.com>

**Pull request merged**